app/models/activity.rb

```diff
@@ -16,9 +16,15 @@ def project=(new_project); end
 
   validates_presence_of :action, :trackable_id, :trackable_type, :user
 
-  VALID_ACTIONS = %w[auto-save create update destroy recover]
-
-  validates_inclusion_of :action, in: VALID_ACTIONS
+  VALID_ACTIONS = {
+    autosave: 'auto-save',
```

Comment:
I had added this here when I was creating activities with auto-save. We're no longer doing that and I wonder if that means I should refactor this back out and put this constant somewhere more appropriate.